hooks/checkout

```diff
@@ -99,19 +102,23 @@ checkout() {
 
   if [[ -z "${BUILDKITE_COMMIT:-}" || "${BUILDKITE_COMMIT}" == "HEAD" ]]; then
     exit_code=0
-    timeout "${GIT_REMOTE_TIMEOUT}" git fetch -v --no-tags origin "${BUILDKITE_BRANCH}" || exit_code=$?
+    echo "COMMAND: GIT_TRACE=1 GIT_CURL_VERBOSE=1 timeout ${GIT_REMOTE_TIMEOUT} git fetch -v --no-tags origin ${BUILDKITE_BRANCH} || exit_code=$?" >> "${git_log}"
+    (GIT_TRACE=1 GIT_CURL_VERBOSE=1 timeout "${GIT_REMOTE_TIMEOUT}" git fetch -v --no-tags origin "${BUILDKITE_BRANCH}" || exit_code=$?) &>> "${git_log}"
```

Comment:
Extract this echoing + logging logic into a function, as it is right now it's super easy to have the two commands drift out of sync.